From 87076b5c498509090b189b57dc2851a98c4d5728 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Tue, 15 Sep 2026 20:06:12 +0200
Subject: [PATCH] PROTO: Fix parallel build race generating the gRPC service
 stubs

Rpc.grpc.pb.cc includes Rpc.pb.h, which in turn includes the headers for
the three protos Rpc.proto imports: File.proto, Recycle.proto and
TrafficShaping.proto. Those are generated by separate custom commands
owned by EosCliProto-Objects, but RpcFileGeneration only depended on
Rpc.pb.{cc,h}, so with enough parallelism EosGrpcProto-Objects could be
compiled first and fail with

  Rpc.pb.h:40:10: fatal error: TrafficShaping.pb.h: No such file or directory

Observed at -j64; a second 'make' always succeeds.
---
 proto/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/proto/CMakeLists.txt b/proto/CMakeLists.txt
index 24a575e59..5ad0b9906 100644
--- a/proto/CMakeLists.txt
+++ b/proto/CMakeLists.txt
@@ -201,7 +201,10 @@ set_target_properties(EosCliProto-Objects PROPERTIES
 #-------------------------------------------------------------------------------
 if (Linux)
      add_custom_target(RpcFileGeneration DEPENDS
-    ${GRPC_SRCS} ${GRPC_HDRS})
+    ${GRPC_SRCS} ${GRPC_HDRS}
+    ${FILE_SRCS} ${FILE_HDRS}
+    ${RECY_SRCS} ${RECY_HDRS}
+    ${TRAFFIC_SHAPING_SRCS} ${TRAFFIC_SHAPING_HDRS})
 
   set(GRPC_PROTOS ${CMAKE_SOURCE_DIR}/common/grpc-proto/Rpc.proto)
   set(GRPC_PROTOBUF_PATH "${CMAKE_BINARY_DIR}/proto/")
